packages/create-cloudflare/src/workers.ts

```diff
@@ -19,27 +19,17 @@ import { detectPackageManager } from "helpers/packages";
 import {
 	chooseAccount,
 	gitCommit,
-	isInsideGitRepo,
 	offerGit,
 	offerToDeploy,
 	printSummary,
 	runDeploy,
-	setupProjectDirectory,
 } from "./common";
-import type { C3Args, C3Context as Context } from "types";
+import type { C3Context } from "types";
 
 const { dlx, npm } = detectPackageManager();
 
-export const runWorkersGenerator = async (args: C3Args) => {
-	const originalCWD = process.cwd();
-	const { name, path } = setupProjectDirectory(args);
-
-	const ctx: Context = {
-		project: { name, path },
-		args,
-		originalCWD,
-		gitRepoAlreadyExisted: await isInsideGitRepo(dirname(path)),
-	};
+export const runWorkersGenerator = async (ctx: C3Context) => {
+	ctx.type = "workers";
```

Comment:
I feel a bit icky about mutating the `ctx` object as it is passed around.
But I see we have been doing that already.

Reply:
I do too but I think I have an idea how we can refactor these steps into middleware that will reduce the ick of these. Stay tuned :D